cli/src/main.rs

```diff
@@ -55,6 +55,29 @@ impl<'a> Tree<'a> {
                             .value_parser(clap::value_parser!(std::net::IpAddr)),
                     ),
 
+                CliCommand::SamlIdentityProviderCreate => cmd
+                    // We're filling in the missing required field so the full
+                    // body is no longer required.
+                    .mut_arg("json-body", |arg| arg.required(false))
+                    .arg(
+                        clap::Arg::new("metadata-url")
+                            .long("metadata-url")
+                            .value_name("url")
+                            .value_parser(clap::value_parser!(String)),
+                    )
+                    .arg(
+                        clap::Arg::new("metadata-value")
+                            .long("metadata-value")
+                            .value_name("xml")
+                            .value_parser(clap::value_parser!(String)),
```

Comment:
Given how long this value can get (it is XML after all), should this take a file instead? If one really wants to pass it as a string they can pipe it to `stdin` with `-` (there's the `<( echo $XML )` redirect syntax but was that just a bash-ism?).

Reply:
so: yeah! however, my primary goal was to make this stuff 1. possible and 2. generateable from progenitor. Could I defer this convenience (which I agree would be much nicer to use).

Reply:
sure! it is an improvement over the status quo